src/android/ContactAccessorSdk5.java

```diff
@@ -307,24 +306,41 @@ public JSONArray search(JSONArray fields, JSONObject options) {
      * @return     a JSONObject representing the contact
      * @throws JSONException
      */
+    @Override
     public JSONObject getContactById(String id) throws JSONException {
-        // Call overloaded version with no desiredFields
-        return getContactById(id, null);
+        Cursor c = mApp.getActivity().getContentResolver().query(
+                ContactsContract.Data.CONTENT_URI,
+                null,
+                ContactsContract.Data.CONTACT_ID + " = ? ",
+                new String[] { id },
+                ContactsContract.Data.CONTACT_ID + " ASC"
+        );
+        return getContactById(c);
     }
 
+    /**
+     * A special search that finds one contact by its raw id
+     *
+     * @param rawId   contact to find by raw id
+     * @return     a JSONObject representing the contact
+     * @throws JSONException
+     */
     @Override
-    public JSONObject getContactById(String id, JSONArray desiredFields) throws JSONException {
-        // Do the id query
+    public JSONObject getContactByRawId(String rawId) throws JSONException {
         Cursor c = mApp.getActivity().getContentResolver().query(
                 ContactsContract.Data.CONTENT_URI,
                 null,
                 ContactsContract.Data.RAW_CONTACT_ID + " = ? ",
-                new String[] { id },
-                ContactsContract.Data.RAW_CONTACT_ID + " ASC");
+                new String[] { rawId },
+                ContactsContract.Data.RAW_CONTACT_ID + " ASC"
+        );
+        return getContactById(c);
+    }
 
+    private JSONObject getContactById(Cursor c) throws JSONException {
         HashMap<String, Boolean> populate = buildPopulationSet(
-                new JSONObject().put("desiredFields", desiredFields)
-                );
+                new JSONObject().put("desiredFields", null)
```

Comment:
same as above

Reply:
We provide null here to create a population set with all attributes - as it was being done before - it wasn't changed here

The search for contacts functionality allows to specify in the javascript which fields do we want to return - this one does not. Although from my tests, the `desiredFields` in search comes as an empty array, so it has the same result.

Reply:
Sorry, what I meant was to ask if we could add a comment in the code explaining why the `null`